### DISCUSSION TITLE
Change looping event types (foreach, repeat, while) to start with an indented subevent

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3649] [Enhancement ideas (and their technical discussions)]
## Description

New users often make a mistake when using ForEach events.  They do one of two things:
-  Make multiple ForEach() loops when they want multiple events to happen.  They don't know they can use subevents under a single ForEach() event.
- They put their events directly below a ForEach (but not indented as subevents) and assume the events would be included as part of the ForEach() event.

## Solution suggested

Automatically indent the first event below all looping event types (foreach, repeat, while) so that is obvious to users what is included in the loop.  Users can create new events below the included event, and those events will be part of the loop.

![UseSubEvents](https://user-images.githubusercontent.com/8879811/113027444-55049000-9147-11eb-8ec5-a9eb65e5113e.png)

**Comment by tristanbob** [comment]
Almost a year later, and I still never put anything in that top event.   :)   (Thanks for converting this to a discussion)

**Comment by Elairyx7301** [comment]
I had a few issues with "for each" loops at the beginning (also with the common 'trigger once inside loops' issue) but after I learned how to handle them I realized that the best way of having multiple loops is to have sub events for one loop instead of multiple loops.

Long story short: I totally agree that this should be turned into a feature as soon as possible so users - doesn't matter if beginner, intermediate or expert - first of all can't get the chance to be confused about the event type and how to use it most efficiently but this change can also save time when creating loops (I should definitely map my key shortcuts a little better for my personal workflow) but adding sub events always takes a couple seconds.

**Comment by tristanbob** [comment]
This is still something that should be implemented.  Here is an example where the user expected the last event to be part of the while loop.

![image](https://user-images.githubusercontent.com/8879811/191811637-29f7dc5b-0c97-4ce5-af7b-76e2a57dfe51.png)

**Comment by arthuro555** [comment]
Instead of auto-indenting that part, which would probably look like a weird forced subevent, what about fully removing the standard action/condition part of non-standard events? That way, we can get rid of the maintenance burden that is to have a lot of similar events code generation in many events, and force the users to put the loops' bodies as subevents getting rid of all ambiguity.

For the while condition, the looping condition is a part of the loop declaration itself, not of the loop body, and isn't comprised of subevents, only conditions, so I would not indent it. IMO that would bring more confusion than it would clarify anything.
